ui/src/ScholarReader.tsx

```diff
@@ -52,6 +53,11 @@ class ScholarReader extends React.PureComponent<ScholarReaderProps, State> {
      * are called from outside ScholarReader.
      */
     this.state = {
+      userId: undefined,
+      userLibrary: null,
+      setUserId: this.setUserId.bind(this),
```

Comment:
don't know if the codebase as the babel plugin for arrow functions to avoid .bind() but if not it's fine as is!